pkg/apiserver/apiserver.go

```diff
@@ -522,3 +535,99 @@ func (app *uploadAPIApp) createAPIService() error {
 	}
 	return nil
 }
+
+func (app *uploadAPIApp) createWebhook() error {
+	dvPathCreate := dvCreateValidatePath
+	uploadtokenPathCreate := uploadtokenCreateValidatePath
+	namespace := util.GetNamespace()
+	registerWebhook := false
+	webhookRegistration, err := app.client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Get(apiWebhookValidator, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			registerWebhook = true
+		} else {
+			return err
+		}
+	}
+
+	webHooks := []admissionregistrationv1beta1.Webhook{
+		{
+			Name: "datavolume-create-validator.cdi.kubevirt.io",
+			Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+				Operations: []admissionregistrationv1beta1.OperationType{
+					admissionregistrationv1beta1.Create,
+				},
+				Rule: admissionregistrationv1beta1.Rule{
+					APIGroups:   []string{datavolumev1alpha1.SchemeGroupVersion.Group},
+					APIVersions: []string{datavolumev1alpha1.SchemeGroupVersion.Version},
+					Resources:   []string{"datavolumes"},
+				},
+			}},
+			ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+				Service: &admissionregistrationv1beta1.ServiceReference{
+					Namespace: namespace,
+					Name:      apiServiceName,
+					Path:      &dvPathCreate,
+				},
+				CABundle: app.signingCertBytes,
+			},
+		},
+		{
+			Name: "uploadtoken-create-validator.cdi.kubevirt.io",
+			Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+				Operations: []admissionregistrationv1beta1.OperationType{
+					admissionregistrationv1beta1.Create,
+				},
+				Rule: admissionregistrationv1beta1.Rule{
+					APIGroups:   []string{uploadv1alpha1.SchemeGroupVersion.Group},
```

Comment:
I mentioned this in the related github issue.  But UploadTokenRequest is not a CRD so it doesn't make sense to have a validating webhook for them.  All UploadTokenRequest references can be deleted.